AirCastingTests/ForgotPasswordTests/ForgotPasswordTest.swift

```diff
@@ -0,0 +1,27 @@
+// Created by Lunar on 16/07/2021.
+//
+
+import XCTest
+@testable import AirCasting
+
+class ForgotPasswordViewModelTest: XCTestCase {
+    
+    private var defaultForgotPasswordViewModel: DefaultForgotPasswordViewModel!
+    
+    override func setUp() {
+        defaultForgotPasswordViewModel = DefaultForgotPasswordViewModel(controller: MockForgotPasswordController())
+    }
+    
+    func test_isEmailChanging() {
+        defaultForgotPasswordViewModel.emailChanged(to: "test@email.com")
+        XCTAssertEqual(defaultForgotPasswordViewModel.email, "test@email.com")
+        XCTAssertNotEqual(defaultForgotPasswordViewModel.email, "@email.com")
```

Comment:
this test is too specific. Namely it's testing an internal implementation of this class which is not really what we want. Note that you had to remove `private` from `var email: String` in the `DefaultForgotPasswordViewModel` class. This is a sign of too specific testing. We want to treat the class under test as a black box - we put X into the input and we expect Y to be the output all using public interfaces.

So for this case we need to test like this:

```Swift
defaultPasswordViewModel.emailChanged(to: "test@gmail.com")
XCTAssertEqual(controllerMock.history.last, "test@gmail.com")
```

Now:
- we supply tested class with an input of our choice (`defaultPasswordViewModel.emailChanged(to: "test@gmail.com")`)
- we catch the output (using an implementation of the `ForgotPasswordController` that is remembering all emails for us).
- we assert that the generated output is indeed what we want.